### PR TITLE
Wildcard algo

### PR DESCRIPTION
**Pull request by Prathameshk2024**
• Matches text strings against patterns with wildcards using dynamic programming
• Supports * (matches any sequence of characters) and ? (matches exactly one character)
• Implements standard DP, space-optimized, and backtracking approaches
• Returns TRUE/FALSE for pattern matching validation
• Includes helper functions for finding and counting matches
• Time complexity: O(m×n), Space: O(m×n) standard or O(n) optimized
• Features 15+ test cases with edge cases, file matching, and complex patterns
• Production-ready implementation for string pattern matching tasks

### CODE REVIEW

**Comment by Prathameshk2024** [comment]
@siriak

File: et --soft HEAD~1 [hunk 1]
```diff
@@ -0,0 +1,461 @@
+[33mcommit 7d4b7af52036b21abf54435f14250ef170351389[m[33m ([m[1;36mHEAD[m[33m -> [m[1;32mGraph_colouring[m[33m)[m
+Author: Prathamesh Kalshetti <prathameshkalshetti450@gmail.com>
+Date:   Sun Oct 19 13:56:38 2025 +0530
```
Comment (siriak) [inline et --soft HEAD~1:3]: This file is invalid

**Review by siriak — CHANGES_REQUESTED** [review]
